### PR TITLE
Carto: fetchMap support for custom markers

### PR DESCRIPTION
**Pull request by zbigg**
#### Background

Carto `fetchMap` shall support custom markers. 

#### Change List
* fetchMap supports custom map markers
* fix handling of radiusField/radiusSize
* new tests for fetchMap

### CODE REVIEW

File: modules/carto/src/api/layer-map.ts [hunk 1]
```diff
@@ -339,6 +340,64 @@ export function getColorAccessor(
   return normalizeAccessor(accessor, data);
 }
 
+const FALLBACK_ICON =
```
Comment (zbigg) [inline modules/carto/src/api/layer-map.ts:343]: This is just filled circle in SVG.

**Review by felixpalmer — CHANGES_REQUESTED** [review]
Nice job :)

File: modules/carto/src/api/layer-map.ts [hunk 2]
```diff
@@ -369,25 +428,31 @@ const FORMATS: Record<string, (value: any) => string> = {
...
-  const calculateOffset = r => [signX * (r + padding), signY * (r + padding + sizeOffset)];
+  const calculateOffset = (r: number): [number, number] => [
+    signX * (r + padding),
+    signY * (r + padding + sizeOffset)
+  ];
 
   return typeof radius === 'function'
-    ? d => {
-        return calculateOffset(radius(d));
+    ? (d, oi) => {
```
[the remark this answers is not in the document]
Comment (zbigg) [reply modules/carto/src/api/layer-map.ts:454]: I was trying to be too type-correct, more than reasonable, accessors are ours and they don't need `oi` removed this excessive typing

File: modules/carto/src/api/parseMap.ts [hunk 3]
```diff
@@ -152,17 +155,42 @@ function createStyleProps(config, mapping) {
...
+    strokeColorScale
+  } = visualChannels;
   let {heightField, heightScale} = visualChannels;
   if (type === 'hexagonId') {
     heightField = sizeField;
     heightScale = sizeScale;
   }
   const {textLabel, visConfig} = config;
-  const result: Record<string, any> = {};
+  let result: Record<string, any> = {};
```
[the remark this answers is not in the document]
Comment (zbigg) [reply modules/carto/src/api/parseMap.ts:193]: OTOH _i_ was unhappy with having too much random stuff in result not necessary needed for this particular mode of layer rendering.
So both are suboptimal :/
I am reverting to original approach.

File: modules/carto/src/api/parseMap.ts [hunk 4]
```diff
@@ -179,44 +207,105 @@ function createChannelProps(visualChannels, type, config, data) {
...
+            }
+          }
+        }
+      }
+    };
+
+    if (getFillColor) {
+      result.getIconColor = getFillColor;
+    } else if (config.color) {
+      result.getIconColor = config.color;
```
[the remark this answers is not in the document]
Comment (zbigg) [reply modules/carto/src/api/parseMap.ts]: [Here](https://github.com/visgl/deck.gl/pull/7250/commits/12df761092256060f6fbfa8490e000b4a38bdc80) I've created customPropMap that is created for layers with customMarkers, let's see if it's looks & feels better

**Review by felixpalmer — APPROVED** [review]

**Review by felixpalmer — APPROVED** [review]

**Pull request merged**